hasher-matcher-actioner/src/OpenMediaMatch/app.py

```diff
@@ -98,6 +106,16 @@ def create_app() -> flask.Flask:
 
     scheduler: APScheduler | None = None
 
+    if _is_gunicorn() and (
+        app.config.get("ROLE_HASHER", False)
+        or app.config.get("ROLE_MATCHER", False)
+        or app.config.get("ROLE_CURATOR", False)
+    ) and (
+        app.config.get("TASK_INDEXER", False)
+        or app.config.get("TASK_FETCHER", False)
+    ):
+        raise RuntimeError("Cannot run Indexer or Fetcher tasks in gunicorn ROLE worker - must be run in separate deployments")
+
```

Comment:
This is to determine the type of deployment? Would it be easier to set and explicit env value via the deployment?

Reply:
Well, workers could have multiple different roles, or there could be different workers for each deploy type, depending on the setup. The key thing is that `TASK_INDEXER` and `TASK_FETCHER` can't be true if any of the ROLE's are true. 

I guess I could set an `IS_WORKER` env var, but I would still want to check for the role vars anyway, so then what's the point? I'm contributing this back to facebook so I want it to be as foolproof as possible